### PR TITLE
[WIP] feat: set loading state on `_app.tsx`

### PR DESCRIPTION
**Pull request by liby**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -43,8 +43,14 @@ jobs:
...
+      - uses: pnpm/action-setup@v2
+        with:
+          version: 7
+          run_install: false
       - uses: actions/setup-node@v3
         with:
-          node-version: '17.x'
+          node-version: '16'
+          cache: 'pnpm'
+      - run: pnpm install --frozen-lockfile
```
Comment (tisonkun) [inline .github/workflows/ci.yml:54]: Why do you make these changes? `setup-node@v3` should already add `pnpm`. Also if we don't have compatibility issues, it's better to keep a newer node version.

File: package.json [hunk 2]
```diff
@@ -2,6 +2,14 @@
   "name": "neptune",
   "version": "0.1.0",
   "private": true,
+  "packageManager": "pnpm@7.6.0",
+  "engines": {
+    "node": ">=16.0.0",
+    "pnpm": ">=7.0.0",
+    "yarn": ">2 <1",
+    "npm": ">2 <1"
```
Comment (tisonkun) [inline package.json:10]: Perhaps stick to pnpm for this project? I don't think this project will take care of multiple package managers.

File: pages/_app.tsx [hunk 3]
```diff
@@ -16,8 +16,33 @@
...
+        };
+        router.events.on("routeChangeStart", start);
+        router.events.on("routeChangeComplete", end);
+        router.events.on("routeChangeError", end);
+        return () => {
+            router.events.off("routeChangeStart", start);
+            router.events.off("routeChangeComplete", end);
+            router.events.off("routeChangeError", end);
+        };
+    }, [router.events]);
```
Comment (tisonkun) [inline pages/_app.tsx:40]: I ever try this trick, but didn't add `[router.events]` part..

Comment (tisonkun) [reply pages/_app.tsx:40]: But from my local env `Loading...` doesn't show.

**Review by tisonkun — COMMENTED** [review]
Thanks for your contribution!

However, I pull this change and run `pnpm dev` locally, but the experience keeps.

File: package.json (hunk 2 above)
Comment (liby) [reply package.json:10]: Same as [only allow pnpm](https://pnpm.io/only-allow-pnpm).
```
"yarn": ">2 <1",
"npm": ">2 <1"
```

Comment (tisonkun) [reply package.json:10]: Then let's explicit said [only allow pnpm](https://pnpm.io/only-allow-pnpm) :)

Comment (liby) [reply package.json:10]: image

File: .github/workflows/ci.yml (hunk 1 above)
Comment (liby) [reply .github/workflows/ci.yml:54]: > Why do you make these changes? setup-node@v3 should already add pnpm. 

[pnpm: command not found](https://github.com/korandoru/neptune/runs/7500230993?check_suite_focus=true)
[setup-node: advanced-usage](https://github.com/actions/setup-node/blob/5b949b50c3461bbcd5a540b150c368278160234a/docs/advanced-usage.md#:~:text=pnpm/action%2Dsetup%40v2)

Comment (liby) [reply .github/workflows/ci.yml:54]: > Also if we don't have compatibility issues, it's better to keep a newer node version.

Sry~ Subconscious habit, and node@16 is Active LTS, so...
> Major Node.js versions enter Current release status for six months, which gives library authors time to add support for them. After six months, odd-numbered releases (9, 11, etc.) become unsupported, and even-numbered releases (10, 12, etc.) move to Active LTS status and are ready for general use. LTS release status is "long-term support", which typically guarantees that critical bugs will be fixed for a total of 30 months. Production applications should only use Active LTS or Maintenance LTS releases.

File: pages/_app.tsx (hunk 3 above)
Comment (liby) [reply pages/_app.tsx:40]: Sigh, it worked briefly, and I now also find that it doesn't seem to work.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (tisonkun) [reply .github/workflows/ci.yml:54]: Oops. Sorry I don't find that CI fails on main branch. Thank you!

File: package.json (hunk 2 above)
Comment (tisonkun) [reply package.json:10]: OK. I don't have a strong feeling on this part. Let's focus on setting loading state here.

Comment (liby) [reply package.json:10]: Maybe consider migrating to `getStaticProps`, and updating or creating content at runtime with Incremental Static Regeneration. Let me go through the doc again.

**Comment by tisonkun** [comment]
Finally I use the original CSR solution to achieve it. You can view the result at https://neptune-bay.vercel.app/affinity/ratio.

**Pull request closed without merging**